### PR TITLE
Auto Reload app

### PR DESCRIPTION
**Pull request by Johngtka**
ezgif com-video-to-gif

### CODE REVIEW

**Comment by Johngtka** [comment]
S&M Commit proposal:
`feat: Refreshing Table when New Patient and Edit`

File: src/app/home/home.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <app-patient-search></app-patient-search>
-<app-patient-table (updatePatient)="openDialog($event)"></app-patient-table>
+<app-patient-table [newOrUpdatedPatient]="newOrUpdatedPatient" (updatePatient)="openDialog($event)"></app-patient-table>
 <button (click)="openDialog()" class="add-btn" mat-mini-fab color="primary">
     <i class="material-icons">how_to_reg</i>
-</button>
+</button>
```
Comment (macrog) [inline src/app/home/home.component.html:5]: missing empty line at the end of the file ❗ do you have prettier? can you make it to format on save please ❗

File: src/app/patient-table/patient-table.component.ts [hunk 2]
```diff
@@ -16,17 +24,39 @@ import {
...
+                    changes['newOrUpdatedPatient'].currentValue;
+                this.dataSource = [...this.dataSource];
+            } else {
+                // new
+                this.dataSource = [
+                    ...this.dataSource,
+                    changes['newOrUpdatedPatient'].currentValue,
+                ];
+            }
+        }
```
Comment (macrog) [inline src/app/patient-table/patient-table.component.ts:58]: do yo understand this part of the code?

**Review by macrog — CHANGES_REQUESTED** [review]

File: src/app/patient-table/patient-table.component.ts (hunk 2 above)
Comment (Johngtka) [reply src/app/patient-table/patient-table.component.ts:58]: not yet but you can explain here

Comment (macrog) [reply src/app/patient-table/patient-table.component.ts:58]: This part is basically searching the index of the patient in the table data set

            const tablePatientIndex = this.dataSource.findIndex(
                (ds: Patient) =>
                    ds._id === changes['newOrUpdatedPatient'].currentValue._id,
            );

If index is !== -1 means that index was found the object exists so we are going to update it

           if (tablePatientIndex !== -1) {
                // update
                this.dataSource[tablePatientIndex] =
                    changes['newOrUpdatedPatient'].currentValue;
                this.dataSource = [...this.dataSource];
            }

other wise we do the insert as it seems to be a new patient added

**Review by macrog — APPROVED** [review]
👍

**Pull request merged**